ggml/src/ggml-metal/ggml-metal.m

```diff
@@ -1875,7 +1891,22 @@ static bool ggml_metal_supports_op(const struct ggml_backend_metal_device_contex
     }
 }
 
-static bool ggml_metal_encode_node(
+static bool nodes_are_same(
```

Comment:
This is the same as `ggml_are_same_layout` which already are duplicated in `ggml-alloc.c` and `ggml-backend.cpp`. Maybe it could be moved as an inline function in `ggml.h`.